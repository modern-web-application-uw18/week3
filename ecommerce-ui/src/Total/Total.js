import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Total extends Component {
    calcTotal = () => {
        const cartItems = this.props.cartItems;
        const sumOfRentals = cartItems.reduce((theSum, item) =>{
            const cost = parseFloat(item.payment.cost);
            return theSum + cost;
        }, 0);
        return sumOfRentals;
    }
    render() {
        return (
            <strong><p className="totalSum">Total: ${this.calcTotal()}</p></strong>
        )
    }
    
}
Total.propTypes = {
    cartItems: PropTypes.array
}
Total.defaultProps = {
    cartItems: []
}
export default Total;