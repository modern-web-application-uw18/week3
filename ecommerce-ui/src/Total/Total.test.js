import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Total from './Total.js';

describe('Total Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Total />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('should render the cart without crashing', ()=>{
        const component = TestRenderer.create(<Total />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

