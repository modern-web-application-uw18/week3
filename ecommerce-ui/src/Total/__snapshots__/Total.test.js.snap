// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Total Component should render the cart without crashing 1`] = `
<strong>
  <p
    className="totalSum"
  >
    Total: $
    0
  </p>
</strong>
`;
